src/react/useSWEffect.ts

```diff
@@ -15,10 +15,37 @@ export function useSWEffect(): void {
     if (typeof p === 'object' && typeof p.then === 'function') {
       return true;
     }
+    return false;
+  }
 
+  function isFunction(p: any): boolean {
+    if(typeof p ==='function') {
+      return true;
+    }
     return false;
   }
 
+
+  function filteredMatches(route) {
```

Comment:
Why are we removing the whole `route` when the `route.data` is a promise? Should we keep the `route` without the `route.data`?

Reply:
We could keep the `route` without the data. I initially presumed that the route won't be needed without data available. But that's wrong